docs/qubit-routing/qaoa-qubit-routing.md

```diff
@@ -0,0 +1,125 @@
+# Qubit Routing for QAOA
+
+OpenQAOA supports custom qubit routing solutions specific to QAOA circuits as part of the QAOA workflow. 
+
+## What is Qubit Routing?
+Qubit Routing is a key step in circuit compilation that maps a Quantum Circuit defined over logical qubits to the physical qubits in a Quantum Computer. Most of currently available Quantum Computers have sparsely connected physical qubits. Therefore, in order to implement an entaglement operation between physical qubits $i, j$ that are not connected on the QPU, we make use of $SWAP$ gates. A SWAP gate is an entangling operation between two qubits $a, b$ that swaps the quantum state of the two qubits. Mathematically, it can be described as follows:
+
+$$
+SWAP (\Psi_a \otimes \Psi_b) = \Psi_b \otimes \Psi_a
+$$
+
+## Why do we need Qubit Routing?
+QAOA circuits are a type of variational circuits with alternating layers of parametrized cost and mixer operations. Typically, the cost layer contains entangling gates between pairs of qubits. In most cases, these entangling operations may act between qubits that are not physically connected on the hardware. In such cases, the circuit compilation uses qubit routing as a solution to map these operations onto the limited connectivity offered by the Quantum Computer. The figure below, as an example, shows these gates marked in red that cannot be implemented due to the lack of an entangling channel between the qubits. 
+
+![qaoa_routing_example_1](/img/routing_1.png) ![routing_example_circuit_2](/img/noswap_ckts.png)
```

Comment:
if possible, I would color in red the gates in the circuit too, to make it more clear.